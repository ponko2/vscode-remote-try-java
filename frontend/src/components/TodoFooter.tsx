import type { LinkProps } from "@tanstack/react-router";
import { Link } from "@tanstack/react-router";
import { cva } from "class-variance-authority";
import clsx from "clsx";
import { TodoButton } from "./TodoButton";

type Props = {
  todosCount: number;
  completedTodosCount: number;
  clearCompletedTodos: () => void;
};

export function TodoFooter({
  todosCount,
  completedTodosCount,
  clearCompletedTodos,
}: Props) {
  if (todosCount <= 0) {
    return null;
  }

  const activeCount = todosCount - completedTodosCount;

  const link = cva(
    [
      "m-1",
      "rounded",
      "border",
      "px-2",
      "py-1",
      "no-underline",
      "hover:border-red-400",
    ],
    {
      variants: {
        intent: {
          active: ["border-red-700"],
          inactive: ["border-transparent"],
        },
      },
    },
  );

  const links: LinkProps[] = [
    { to: "/", children: "All" },
    { to: "/active", children: "Active" },
    { to: "/completed", children: "Completed" },
  ];

  return (
    <footer
      className={clsx(
        "isolate",
        "grid",
        "grid-cols-2",
        "gap-2",
        "px-4",
        "py-2.5",
        "sm:grid-cols-3",
      )}
    >
      <span>
        <strong className="font-light">{activeCount ?? "No"}</strong>{" "}
        {activeCount === 1 ? "item" : "items"} left
      </span>
      <ul
        className={clsx(
          "order-last",
          "col-span-full",
          "text-center",
          "sm:order-none",
          "sm:col-auto",
        )}
      >
        {links.map(({ to, children }) => (
          <li className="inline" key={to}>
            <Link
              activeProps={{ className: link({ intent: "active" }) }}
              inactiveProps={{ className: link({ intent: "inactive" }) }}
              to={to}
            >
              {children}
            </Link>
          </li>
        ))}
      </ul>
      {!!completedTodosCount && (
        <TodoButton
          className={clsx(
            "cursor-pointer",
            "text-right",
            "no-underline",
            "hover:underline",
            "active:no-underline",
          )}
          onClick={clearCompletedTodos}
        >
          Clear completed
        </TodoButton>
      )}
    </footer>
  );
}
