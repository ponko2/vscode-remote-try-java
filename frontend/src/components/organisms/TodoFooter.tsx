import { Link } from "@tanstack/react-location";
import { TodoButton } from "../../components/atoms/TodoButton";
import styles from "./TodoFooter.module.css";

type Props = {
  todosCount: number;
  completedCount: number;
  clearCompleted: () => void;
};

function getActiveProps() {
  return { className: [styles.selected] };
}

export const TodoFooter = ({
  todosCount,
  completedCount,
  clearCompleted,
}: Props): JSX.Element | null => {
  if (todosCount <= 0) {
    return null;
  }

  const activeCount = todosCount - completedCount;

  return (
    <footer className={styles.footer}>
      <span className={styles.todoCount}>
        <strong>{activeCount ?? "No"}</strong>{" "}
        {activeCount === 1 ? "item" : "items"} left
      </span>
      <ul className={styles.filters}>
        <li>
          <Link to="/" className={styles.link} getActiveProps={getActiveProps}>
            All
          </Link>
        </li>
        <li>
          <Link
            to="active"
            className={styles.link}
            getActiveProps={getActiveProps}
          >
            Active
          </Link>
        </li>
        <li>
          <Link
            to="completed"
            className={styles.link}
            getActiveProps={getActiveProps}
          >
            Completed
          </Link>
        </li>
      </ul>
      {!!completedCount && (
        <TodoButton className={styles.clearCompleted} onClick={clearCompleted}>
          Clear completed
        </TodoButton>
      )}
    </footer>
  );
};
