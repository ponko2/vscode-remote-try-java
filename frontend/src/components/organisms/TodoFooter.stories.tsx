import { action } from "@storybook/addon-actions";
import type { Meta, StoryFn } from "@storybook/react";
import {
  createMemoryHistory,
  createRouteConfig,
  ReactRouter,
  RouterProvider,
} from "@tanstack/react-router";
import { TodoFooter } from "./TodoFooter";

const history = createMemoryHistory({ initialEntries: ["/"] });

export default {
  component: TodoFooter,
  decorators: [
    (Story) => {
      const rootRoute = createRouteConfig({ component: () => <Story /> });
      const routeConfig = rootRoute.addChildren([
        rootRoute.createRoute({ path: "/" }),
        rootRoute.createRoute({ path: "/active" }),
        rootRoute.createRoute({ path: "/completed" }),
      ]);
      const router = new ReactRouter({ routeConfig, history });
      return <RouterProvider router={router} />;
    },
  ],
} as Meta<typeof TodoFooter>;

const Template: StoryFn<typeof TodoFooter> = (args) => (
  <TodoFooter {...args} clearCompleted={action("clearCompleted")} />
);

export const Basic = Template.bind({});

Basic.args = {
  todosCount: 1,
  completedCount: 1,
};

export const HasCompleted = Template.bind({});

HasCompleted.args = {
  todosCount: 2,
  completedCount: 1,
};
