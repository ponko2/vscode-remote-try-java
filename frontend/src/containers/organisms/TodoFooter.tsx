import { TodoFooter as Component } from "../../components/organisms/TodoFooter";
import {
  useClearCompleted,
  useCompletedCount,
  useTodosCount,
} from "../../hooks/todos";

export const TodoFooter = (): JSX.Element => {
  const { data: todosCount = 0 } = useTodosCount();
  const { data: completedCount = 0 } = useCompletedCount();
  const clearCompleted = useClearCompleted();
  return (
    <Component
      todosCount={todosCount}
      completedCount={completedCount}
      clearCompleted={clearCompleted.mutate}
    ></Component>
  );
};
