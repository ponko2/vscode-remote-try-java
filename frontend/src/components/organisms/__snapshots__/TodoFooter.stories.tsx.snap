// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/organisms/TodoFooter Basic smoke-test 1`] = `
<footer class="_footer_1k0lb_1">
  <span class="_todoCount_1k0lb_28">
    <strong>
      0
    </strong>
    items left
  </span>
  <ul class="_filters_1k0lb_37">
    <li>
      <a class="_link_1k0lb_69 _selected_1k0lb_82"
         href="/"
         data-status="active"
      >
        All
      </a>
    </li>
    <li>
      <a href="/active"
         class="_link_1k0lb_69"
      >
        Active
      </a>
    </li>
    <li>
      <a href="/completed"
         class="_link_1k0lb_69"
      >
        Completed
      </a>
    </li>
  </ul>
  <button class="_button_z78gq_1 _clearCompleted_1k0lb_56">
    Clear completed
  </button>
</footer>
`;

exports[`components/organisms/TodoFooter HasCompleted smoke-test 1`] = `
<footer class="_footer_1k0lb_1">
  <span class="_todoCount_1k0lb_28">
    <strong>
      1
    </strong>
    item left
  </span>
  <ul class="_filters_1k0lb_37">
    <li>
      <a class="_link_1k0lb_69 _selected_1k0lb_82"
         href="/"
         data-status="active"
      >
        All
      </a>
    </li>
    <li>
      <a href="/active"
         class="_link_1k0lb_69"
      >
        Active
      </a>
    </li>
    <li>
      <a href="/completed"
         class="_link_1k0lb_69"
      >
        Completed
      </a>
    </li>
  </ul>
  <button class="_button_z78gq_1 _clearCompleted_1k0lb_56">
    Clear completed
  </button>
</footer>
`;
